enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.ts

```diff
@@ -19,9 +19,77 @@ import type {
   SdkIframeEmbedTagMessage,
   SdkIframeEmbedTagSettings,
 } from "./types/embed";
+import { attributeToSettingKey, parseAttributeValue } from "./webcomponents";
 
 const EMBEDDING_ROUTE = "embed/sdk/v1";
 
+/** list of active embeds, used to know which embeds to update when the global config changes */
+const _activeEmbeds: Set<MetabaseEmbedElement> = new Set();
+
+// Stub of MetabaseEmbedElement to satisfy type requirements of helper utilities below.
+// The full custom elements are declared later in this file.
+export class MetabaseEmbedElement extends HTMLElement {
+  updateSettings(_settings: Partial<SdkIframeEmbedSettings>) {}
+}
+
+// Setup a proxy to watch for changes to window.metabaseConfig and update all
+// active embeds when the config changes. It also setups a setter for
+// window.metabaseConfig to re-create the proxy if the whole object is replaced,
+// for example if this script is loaded before the customer calls
+// `defineMetabaseConfig` in their code, which replaces the entire object.
```

Comment:
If the script is loaded _after_ they call `defineMetabaseConfig`, then it still goes through the createProxy to update underlying embeds, right?